src/EditResultsTable.re

```diff
@@ -122,133 +125,151 @@ let make = (~results: list(result), ~communityName: string, ~queryToRefetch) =>
     };
   };
 
-  <Paper>
-    <div className="title">
-      <Typography variant="h6"> {text("Results")} </Typography>
-    </div>
-    <Table size="small">
-      <TableHead>
-        <TableRow>
-          <TableCell> {text("Edit")} </TableCell>
-          <TableCell align="right"> {text("Player1")} </TableCell>
-          <TableCell style=numberCellStyle> {text("G1")} </TableCell>
-          <TableCell style=colonStyle />
-          <TableCell style=numberCellStyle> {text("G2")} </TableCell>
-          <TableCell> {text("Player2")} </TableCell>
-          <TableCell style=extraTimeStyle align="right" title="Extra time">
-            {text("E")}
-          </TableCell>
-          <TableCell style=dateStyle> {text("Date")} </TableCell>
-        </TableRow>
-      </TableHead>
-      <TableBody>
-        {results
-         ->Belt.List.map(result =>
-             <TableRow key={string_of_int(result.id)}>
-               <TableCell>
+  switch (settingsQuery) {
+  | Loading => <CircularProgress />
+  | NoData
+  | Error(_) => <span> {text("Error")} </span>
+  | Data(communitySettings) =>
+    <Paper>
+      <div className="title">
+        <Typography variant="h6"> {text("Results")} </Typography>
+      </div>
+      <Table size="small">
+        <TableHead>
+          <TableRow>
+            <TableCell> {text("Edit")} </TableCell>
+            <TableCell align="right"> {text("Player1")} </TableCell>
+            <TableCell style=numberCellStyle> {text("G1")} </TableCell>
+            <TableCell style=colonStyle />
+            <TableCell style=numberCellStyle> {text("G2")} </TableCell>
+            <TableCell> {text("Player2")} </TableCell>
+            <TableCell style=extraTimeStyle align="right" title="Extra time">
+              {text("E")}
+            </TableCell>
+            <TableCell style=dateStyle> {text("Date")} </TableCell>
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {results
+           ->Belt.List.map(result =>
+               <TableRow key={string_of_int(result.id)}>
+                 <TableCell>
+                   {switch (state) {
+                    | NotEditing =>
+                      <button onClick={_ => dispatch(StartEditing(result))}>
+                        {text("Edit")}
+                      </button>
+                    | Editing(editedResult) =>
```

Comment:
I don't understand why `git` is going crazy here, it is the same code, just indented deeper ...

Reply:
Indeed...